### PR TITLE
Added way to resolve several embedded @contexts in a json payload

### LINKED ISSUE #183: Resolving context is not recursive
The method [expand](https://github.com/BlueBrain/nexus-kg/blob/8c5e8045ff862d4ca2998aa2a1f96b419fe20413/modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/contexts/Contexts.scala#L217) in `Contexts` only resolves the top context.

What about resolving every `@context`in the objects that might be embedded?

For example that would not work now:

```json
{
  "@context": [
    "{{base}}/contexts/nexus/core/standards/v0.1.0",
    {
      "@base": "https://nexus.example.com/dev/v0/voc/experiment/",
      "@vocab": "https://nexus.example.com/vocabs/bbp/neurosciencegraph/core/v0.1.0/",
      "familyName": "http://schema.org/familyName",
      "givenName": "http://schema.org/givenName",
      "prov": "http://www.w3.org/ns/prov#",
      "schema": "http://schema.org/",
      "links": {
        "@id": "nxv:links"
      }
    }
  ],
  "@id": "https://nexus.example.com/dev/v0/data/bbp/core/person/v0.1.0/44b97c0d-fe24-4c9a-9c01-3e5feb16543d",
  "@type": "schema:Person",
  "familyName": "Maria",
  "givenName": "Toledo",
  "nxv:rev": 1,
  "nxv:deprecated": false,
  "links": {
    "@context": "{{base}}/contexts/nexus/core/links/v0.1.0",
    "self": "https://nexus.example.com/dev/v0/data/bbp/core/person/v0.1.0/44b97c0d-fe24-4c9a-9c01-3e5feb16543d"
  }
}
```

Only the first context will be resolved while the one in `links` it won't.

### PR DESCRIPTION
**Pull request by umbreak**
Fixes #183

### CODE REVIEW

**Comment by umbreak** [comment]
retest this please

**Comment by umbreak** [comment]
retest this please

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/contexts/Contexts.scala [hunk 1]
```diff
@@ -269,16 +266,24 @@ final class Contexts[F[_]](agg: ContextAggregate[F], doms: Domains[F], baseUri:
...
-        case (_, _, Some(_)) => F.pure(value)
+        case (_, _, Some(_)) => F.pure(context)
         case (_, _, _)       => F.raiseError(shapeValidationFailure)
       }
 
-    inner(context).map(json => context deepMerge Json.obj("@context" -> json))
+    def inner(jsonObj: JsonObject): F[JsonObject] =
+      F.sequence(jsonObj.toVector.map {
+        case ("@context", v) => resolveValue(v).map("@context" -> _)
+        case (k, v)          => resolve(v).map(k               -> _)
```
Comment (wwajerowicz) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/contexts/Contexts.scala]: This formatting looks a bit weird.

Comment (umbreak) [reply modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/contexts/Contexts.scala]: It was scalafmt one...but yeah looks a bit weird

**Pull request merged**